Shopping List/ShoppingCollectionViewController.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  ShoppingCollectionViewController.swift
+//  Shopping List
+//
+//  Created by Cameron Dunn on 1/18/19.
+//  Copyright © 2019 Lambda School. All rights reserved.
+//
+
+import UIKit
+
+private let reuseIdentifier = "Cell"
+
+class ShoppingCollectionViewController: UICollectionViewController, ButtonDelegate {
+    func tappedButton(on: CustomCollectionViewCell) {
+        model.editItemInList(sender: on)
+        model.saveList()
+        collectionView?.reloadData()
+    }
+    let model = ShoppingListModel()
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if(segue.identifier == "NextSegue"){
+            model.createShoppingListArray()
+            let destination = segue.destination as! CheckoutViewController
+            destination.model = model
+        }
+    }
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        model.createShoppingList()
+        model.loadList()
+        
+        // Uncomment the following line to preserve selection between presentations
+        // self.clearsSelectionOnViewWillAppear = false
+
+        // Register cell classes
+        self.collectionView!.register(UICollectionViewCell.self, forCellWithReuseIdentifier: reuseIdentifier)
+
+        // Do any additional setup after loading the view.
+    }
+
+    /*
+    // MARK: - Navigation
+
+    // In a storyboard-based application, you will often want to do a little preparation before navigation
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        // Get the new view controller using [segue destinationViewController].
+        // Pass the selected object to the new view controller.
+    }
+    */
+
+    // MARK: UICollectionViewDataSource
+
+    override func numberOfSections(in collectionView: UICollectionView) -> Int {
+        // #warning Incomplete implementation, return the number of sections
+        return 1
+    }
+
+
+    override func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        // #warning Incomplete implementation, return the number of items
+        return model.shoppingList.count
+    }
+
+    override func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "CustomCell", for: indexPath) as! CustomCollectionViewCell
+        cell.label.text = model.shoppingList[indexPath.row].nameOfItem
+        cell.imageView.image = UIImage(data: model.shoppingList[indexPath.row].image!)
+        cell.item = model.shoppingList[indexPath.row]
+        cell.delegate = self
+        if(model.shoppingList[indexPath.row].itemAdded == false){
+            cell.button.setTitle("Not Added", for: .normal)
```

Comment:
Line 67-68-69 and 71-75. All of these should move to `CustomCollectionViewCell`. If you check `Planets` guided project, which we did in the class, you will learn how to implement `cellForItemAt indexPath` and pass data to a custom cell class.